packages/sdk-react/src/components/receivables/InvoiceDetails/ExistingInvoiceDetails/components/EditInvoiceDetails.tsx

```diff
@@ -0,0 +1,287 @@
+              color="primary"
+              type="submit"
+              form="receivablesDetailsForm"
+              disabled={isLoading}
+            >{t(i18n)`Update`}</Button>
+          </Box>
+        </Toolbar>
+      </DialogTitle>
+      <Divider />
+      <DialogContent>
+        <FormProvider {...methods}>
+          <form
+            id="receivablesDetailsForm"
+            noValidate
+            onSubmit={handleSubmit((values) => {
+              const lineItems: UpdateLineItems = {
+                data: values.line_items.map((lineItem) => ({
+                  quantity: lineItem.quantity,
+                  product_id: lineItem.product_id,
+                  vat_rate_id: lineItem.vat_rate_id,
+                })),
+              };
+
+              const invoicePayload: ReceivableUpdatePayload = {
+                invoice: {
+                  /** Customer section */
+                  counterpart_id: values.counterpart_id,
+                  counterpart_vat_id_id: values.counterpart_vat_id_id,
+                  currency: actualCurrency,
+                  vat_exemption_rationale: values.vat_exemption_rationale,
+                  /**
+                   * Note: We shouldn't send `counterpart_billing_address`
+                   *  because it's auto-selected from UI.
+                   * There is no way to change it (at least right now)
+                   */
+                  counterpart_shipping_address: counterpartShippingAddress
+                    ? {
+                        country: counterpartShippingAddress.country,
+                        city: counterpartShippingAddress.city,
+                        postal_code: counterpartShippingAddress.postal_code,
+                        state: counterpartShippingAddress.state,
+                        line1: counterpartShippingAddress.line1,
+                        line2: counterpartShippingAddress.line2,
+                      }
+                    : undefined,
+                  /** We shouldn't send an empty string to the server if the value is not set */
+                  entity_bank_account_id:
+                    values.entity_bank_account_id || undefined,
+                  payment_terms_id: values.payment_terms_id ?? '',
+                  entity_vat_id_id: values.entity_vat_id_id || undefined,
+                  fulfillment_date: values.fulfillment_date
+                    ? /**
+                       * We have to change the date as Backend accepts it.
+                       * There is no `time` in request, only year, month and date
+                       */
+                      format(values.fulfillment_date, 'yyyy-MM-dd')
+                    : undefined,
+                  /** !!! Note !!! Backend is not supported to edit `purchase_order` so we have to remove it */
+                  // purchase_order: values.purchase_order || undefined,
+                },
+              };
+
+              updateReceivableLineItems.mutate(lineItems, {
```

Comment:
If possible do not update `lineItems` if not changed, that would be cool.

Reply:
Yep, it sounds good. But do you know how to check it? Because as I know `react-hook-form` will create a new array of `line_items` and if we check `current.lineItems === server.lineItems` we will have always `false` because it will be a new array.

So the only one way is to check it by deep check (check all fields by going through all items) which I think will be worse than current one.

Maybe if we could handle `isDirty` but not for the whole form but only for `line_items` we could avoid any checks but I didn't find any options to do so